projects/packages/forms/src/blocks/contact-form/edit.js

```diff
@@ -222,14 +215,6 @@ function JetpackContactFormEdit( { name, attributes, setAttributes, clientId, cl
 					{ ! isSimpleSite() && canUserInstallPlugins && (
 						<IntegrationControls attributes={ attributes } setAttributes={ setAttributes } />
 					) }
-
-					{ isSalesForceExtensionEnabled && salesforceData?.sendToSalesforce && (
-						<SalesforceLeadFormSettings
-							salesforceData={ salesforceData }
-							setAttributes={ setAttributes }
-							instanceId={ instanceId }
-						/>
-					) }
```

Comment:
Since Salesforce settings are managed via the integrations modal, we no longer need this sidebar panel. I have not yet deleted the actual component code for this. I was going to wait until we release this and have it out for a bit first.